packages/vscode-extension/src/commands/UploadFile.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ ** Copyright (c) 2020 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+
+import * as path from 'path';
+import { window, workspace, InputBoxOptions } from 'vscode';
+import SuiteCloudRunner from '../core/SuiteCloudRunner';
+import MessageService from '../service/MessageService';
+import { COMMAND, UPLOAD_FILE, YES, NO } from '../service/TranslationKeys';
+import { actionResultStatus, CLIConfigurationService, ApplicationConstants, FileCabinetService } from '../util/ExtensionUtil';
+import BaseAction from './BaseAction';
+
+export default class UploadFile extends BaseAction {
+	readonly commandName: string = 'file:upload';
+
+	async execute(opts: { suiteCloudRunner: SuiteCloudRunner; messageService: MessageService }) {
+		const activeFile = window.activeTextEditor?.document.uri;
+		if (!activeFile) {
+			opts.messageService.showCommandError(this.translationService.getMessage(UPLOAD_FILE.ERRORS.INVALID_ACTIVE_FILE));
+			return;
+		}
+		const workspaceFolder = workspace.getWorkspaceFolder(activeFile);
+		if (!workspaceFolder) {
+			opts.messageService.showCommandError(this.translationService.getMessage(UPLOAD_FILE.ERRORS.INVALID_WORKSPACE));
+			return;
+		}
+
+		const cliConfigurationService = new CLIConfigurationService();
+		cliConfigurationService.initialize(workspaceFolder.uri.fsPath);
+		cliConfigurationService.getProjectFolder(this.commandName);
+		const projectFolder = cliConfigurationService.getProjectFolder();
```

Comment:
One question...Why do we repeat two times getProjectFolder? Maybe one of them could be deleted.
(I have been analyzing a little bit the code and I think first one is ok, but maybe there is something I missed)